gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsBatchTest.java

```diff
@@ -0,0 +1,649 @@
+    try {
+      batchResult.get();
+      fail("Should throw a DnsException on error.");
+    } catch (DnsException ex) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testGetChangeRequestWithOptions() {
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock);
+    EasyMock.replay(dns);
+    EasyMock.reset(batchMock);
+    EasyMock.reset(optionsMock);
+    Capture<RpcBatch.Callback<Change>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addGetChangeRequest(EasyMock.eq(ZONE_NAME),
+        EasyMock.eq(CHANGE_REQUEST_COMPLETE.generatedId()), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<ChangeRequest> batchResult = dnsBatch.getChangeRequest(ZONE_NAME,
+        CHANGE_REQUEST_COMPLETE.generatedId(), CHANGE_GET_FIELDS);
+    assertNotNull(callback.getValue());
+    String selector = (String) capturedOptions.getValue().get(CHANGE_GET_FIELDS.rpcOption());
+    assertTrue(selector.contains(Dns.ChangeRequestField.STATUS.selector()));
+    assertTrue(selector.contains(Dns.ChangeRequestField.ID.selector()));
+    RpcBatch.Callback<Change> capturedCallback = callback.getValue();
+    EasyMock.expect(optionsMock.service()).andReturn(dns);
+    EasyMock.replay(optionsMock);
+    capturedCallback.onSuccess(CHANGE_REQUEST_COMPLETE.toPb());
+    assertEquals(CHANGE_REQUEST_COMPLETE.toPb(), batchResult.get().toPb());
+  }
+
+  @Test
+  public void testApplyChangeRequest() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Change>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addApplyChangeRequest(EasyMock.eq(ZONE_NAME),
+        EasyMock.eq(CHANGE_REQUEST_PARTIAL.toPb()), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<ChangeRequest> batchResult = dnsBatch.applyChangeRequest(ZONE_INFO.name(),
+        CHANGE_REQUEST_PARTIAL);
+    assertEquals(0, capturedOptions.getValue().size());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<Change> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a DnsException on error.");
+    } catch (DnsException ex) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testApplyChangeRequestWithOptions() {
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock);
+    EasyMock.replay(dns);
+    EasyMock.reset(batchMock);
```

Comment:
Nit: EasyMock methods regarding mock objects take varargs in most cases.  As a result, you can combine this line and the line below into `EasyMock.reset(batchMock, optionsMock);` here.  The same applies elsewhere in this file.